### ISSUE TITLE
protectedts,sql: use protected timestamps to protect index backfill

### ISSUE THREAD

**Issue by ajwerner** [cockroachdb/cockroach#54782]
**Is your feature request related to a problem? Please describe.**

Protected timestamps were introduced in the 20.1 development cycle. The subsystem allows jobs to protect data from being garbage collected. These protected timestamps were applied to backups and changefeeds but not to schema changes. This needs to be remedied as missing protected timestamps can be catastrophic as an index backfill attempts to read historical data.

Jira issue: CRDB-3718

**Comment by ajwerner** [comment]
I don't think this matters anymore really until we lower the gc ttl dramatically. If we do, then we'll want some form of this to avoid having to restart the backfill a bunch. Given the dependencies: cold storage.

**Comment by fqazi** [comment]
Closing in favor of https://github.com/cockroachdb/cockroach/issues/140629
